tests/gpu/CMakeLists.txt

```diff
@@ -214,6 +214,14 @@ elseif(PROTEUS_ENABLE_CUDA)
     endif()
 endif()
 
+if(PROTEUS_ENABLE_HIP)
+SET(lang HIP)
+else()
+SET(lang CUDA)
+endif()
+
+
+
```

Comment:
Why this?

Reply:
Makes sense, let's move this definition at top-level and remove re-setting it within individual functions